Assignment4/martatubia-071350079/task08.py

```diff
@@ -0,0 +1,86 @@
+# -*- coding: utf-8 -*-
+"""Task08.ipynb
+
+Automatically generated by Colaboratory.
+
+Original file is located at
+    https://colab.research.google.com/drive/1TSCslMeVVBLn5Bm6l12axrM_w1z3czHs
+
+**Task 08: Completing missing data**
+"""
+
+!pip install rdflib
+github_storage = "https://raw.githubusercontent.com/FacultadInformatica-LinkedData/Curso2021-2022/master/Assignment4/course_materials"
+
+from rdflib import Graph, Namespace, Literal, URIRef
+from rdflib.namespace import RDF, RDFS
+g1 = Graph()
+g2 = Graph()
+g1.parse(github_storage+"/rdf/data01.rdf", format="xml")
+g2.parse(github_storage+"/rdf/data02.rdf", format="xml")
+
+for s,r,o in g1:
+  print(s,r,o)
+
+for s,r,o in g2:
+  print(s,r,o)
+
+"""Tarea: lista todos los elementos de la clase Person en el primer grafo (data01.rdf) y completa los campos (given name, family name y email) que puedan faltar con los datos del segundo grafo (data02.rdf). Puedes usar consultas SPARQL o iterar el grafo, o ambas cosas."""
+
+from rdflib.plugins.sparql import prepareQuery
+vcard= Namespace("http://www.w3.org/2001/vcard-rdf/3.0#")
+q = prepareQuery('''
+  SELECT 
+    ?s ?p ?o
+  WHERE { 
+    ?s vcard:Given ?o
+    
+  }
+  ''',
+  initNs = {"vcard": vcard}
+)
+
+
+
+for r in g2.query(q):
```

Comment:
this would add all the elements in g2 with given name in g1. But you are asked to complete only the persons in g1 from the info in g2. So you have to search for those persons first.